### PR TITLE
feat: btc fee collection

### PR DESCRIPTION
**Pull request by tcar121293**
btc fee collection
## Related Issue Or Context

Closes: #290 

## How Has This Been Tested? Testing details.
e2e tests + unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/btc/listener/util.go [hunk 1]
```diff
@@ -6,15 +6,18 @@ import (
...
+	"github.com/btcsuite/btcd/btcutil"
 )
 
 const (
 	WitnessV1Taproot = "witness_v1_taproot"
 	OP_RETURN        = "nulldata"
 )
 
-func DecodeDepositEvent(evt btcjson.TxRawResult, resource config.Resource) (Deposit, bool, error) {
+func DecodeDepositEvent(evt btcjson.TxRawResult, resource config.Resource, feeAddress btcutil.Address) (Deposit, bool, error) {
```
Comment (mpetrun5) [inline chains/btc/listener/util.go:17]: Btw how do we handle when there might be multiple deposit events inside one transaction?

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

File: chains/btc/listener/util.go (hunk 1 above)
Comment (tcar121293) [reply chains/btc/listener/util.go:17]: With this implementation, it won't work. The last OP_RETURN evm address will get all the deposits. If we have a standard of how the deposit looks in terms of order of output data eg. 1 output amount, second fee, third destination. Then we could handle multiple deposits in one transaction. I can open an issue for that, so we can handle this in a separate PR

Comment (mpetrun5) [reply chains/btc/listener/util.go:17]: Let's make an issue for this to track it

Comment (tcar121293) [reply chains/btc/listener/util.go:17]: Here is the [issue](https://app.zenhub.com/workspaces/sygma-protocol-62f126a54b7d5a55203732f0/issues/gh/sygmaprotocol/sygma-relayer/302)

**Pull request merged**